datanode/snapshot/config.go

```diff
@@ -0,0 +1,24 @@
+package snapshot
+
+import "code.vegaprotocol.io/vega/datanode/config/encoding"
+
+type Config struct {
+	RemoveSnapshotsOnStartup encoding.Bool `long:"remove-snapshops-on-startup" description:"if true the nodes snapshot directory will be emptied at startup"`
+
+	BlockInterval int64         `long:"block-interval" description:"the block interval between creation of snapshots, 0 if snapshot create should be disabled (default: 0)"`
+	Publish       encoding.Bool `long:"publish" description:"set to true if this node should publish snapshot data"`
+	PanicOnError  encoding.Bool `long:"panic-on-error" description:"true if service should panic on snapshot creation error"`
+
+	DatabaseSnapshotsPath string `long:"database-snapshot-path" description:"the snapshots path relative the database working directory"`
+}
+
+// NewDefaultConfig creates an instance of the package specific configuration, given a
+// pointer to a logger instance to be used for logging within the package.
+func NewDefaultConfig() Config {
+	return Config{
+		DatabaseSnapshotsPath:    "/snapshots",
+		RemoveSnapshotsOnStartup: false,
+		BlockInterval:            1000, // Roughly every 15 mins at 1 block per second
+		Publish:                  false,
```

Comment:
Shouldn't this default to true? By default I'd expect all datanode to be "nice player" and publish every snapshots they do.

Reply:
its a work in progress, once the publisher is done, yes, true would be the default, as the moment I've just hooked the publisher in, its not doing anything yet.